daffodil-core/src/main/scala/org/apache/daffodil/dsom/InitiatedTerminatedMixin.scala

```diff
@@ -61,7 +64,23 @@ trait InitiatedTerminatedMixin
     res
   }
 
-  private lazy val initiatorDiscriminator = prod("initiatorDiscriminator", parentSaysInitiatedContent) { InitiatedContent(this) }
+  private lazy val initiatorDiscriminator = prod(
+    "initiatorDiscriminator",
+    parentSaysInitiatedContent &&
+      immediatelyEnclosingModelGroup.map {
+        case c: ChoiceTermBase => true
+        case s: SequenceTermBase => (isArray || isOptional) &&
+          isVariableOccurrences
+      }.getOrElse(false)) {
```

Comment:
Would it be possible to put this guard in an variable? It wasn't immediately clear to me that this is where the guard ended and the grammar stared.